### PR TITLE
Feature/231 upstage command line argument parsing

### PR DESCRIPTION
**Pull request by gremh97**
## Purpose

* Implements command-line argument parsing support for Upstage connector as requested in Issue #231
* Adds CLI arguments (`--base-url`, `--api-key`, `--model`) to configure Upstage connector settings at runtime
* Establishes configuration priority system: CLI Arguments > Environment Variables > Configuration Files
* Provides comprehensive test coverage with 24 unit tests to ensure robust argument parsing functionality
* Maintains backward compatibility - purely additive feature with no breaking changes

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] New feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## README updated?

The top-level readme for this repo contains a link to each sample in the repo. If you're adding a new sample did you update the readme?

```
[ ] Yes
[ ] No
[x] N/A
```

## How to Test
*  Get the code

```
git clone https://github.com/gremh97/open-chat-playground.git
cd open-chat-playground
git checkout feature/231-upstage-command-line-argument-parsing
```

* Test the code

```
# Run all Upstage argument option tests
dotnet test --filter "UpstageArgumentOptionsTests"

# Run specific test categories
dotnet test --filter "Category=UnitTest"

# Test CLI arguments manually
dotnet run -- --base-url "https://api.upstage.ai/v1" --api-key "your-key" --model "solar-mini"

# Run all tests to ensure no regressions
dotnet test

# Test specific scenarios with detailed output
dotnet test --filter "Given_CLI_BaseUrl_When_Parse_Invoked_Then_It_Should_Use_CLI_BaseUrl" --logger "console;verbosity=detailed"

# Test configuration priority
export UPSTAGE_API_KEY="env-key"
dotnet run -- --api-key "cli-key"  # CLI should override environment variable
```

## What to Check
Verify that the following are valid
* All 24 UpstageArgumentOptionsTests pass successfully (0 failed, 0 skipped)
* CLI arguments `--base-url`, `--api-key`, `--model` work correctly for Upstage connector
* Configuration priority system enforced: CLI > Environment Variables > Config Files
* Help system displays appropriate messages for unknown/invalid arguments
* Edge cases handled properly: missing values, unknown flags, special characters in model names
* No breaking changes to existing connector functionality
* New files follow project coding conventions and XML documentation standards
* Integration with existing ArgumentOptions.Parse() method works seamlessly
* Upstage connector type properly recognized in ConnectorType enum
* Default configuration values loaded correctly from appsettings.json

### Files Added:
- **`src/OpenChat.PlaygroundApp/Options/UpstageArgumentOptions.cs`** - New CLI argument parsing implementation for Upstage
- **`test/OpenChat.PlaygroundApp.Tests/Options/UpstageArgumentOptionsTests.cs`** - Comprehensive test suite with 24 test cases

### Files Modified:
- **`src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs`** - Added Upstage case to connector switch statement (lines 155-160)
- **`src/OpenChat.PlaygroundApp/appsettings.json`** - Added Upstage default configuration section (lines 57-61)

### Test Coverage Breakdown:
- **Basic CLI Parsing**: 4 tests (individual arguments + combined)
- **Configuration Priority**: 6 tests (CLI > Env > Config validation)
- **Edge Cases**: 6 tests (missing values, unknown args, special chars)
- **Help System Integration**: 8 tests (help flag behavior scenarios)

### Configuration Priority Example:
```bash
# 1. Config file (lowest priority)
# appsettings.json: "BaseUrl": "https://api.upstage.ai/v1"

# 2. Environment variables (medium priority)
export UPSTAGE_BASE_URL="https://env.upstage.ai/v1"

# 3. CLI arguments (highest priority)
dotnet run -- --base-url "https://cli.upstage.ai/v1"
# Result: CLI value wins
```

### Supported CLI Arguments:
```bash
--base-url "https://api.upstage.ai/v1"    # Upstage API endpoint
--api-key "sk-your-api-key"               # Upstage API authentication key  
--model "solar-mini"                      # Upstage model name (e.g., solar-mini, solar-1-mini)
```

### Implementation Notes:
- Follows existing patterns from other connectors (OpenAI, Azure AI, etc.)
- Uses Given-When-Then test naming convention for clarity
- Comprehensive XML documentation for all public members
- Handles nullable string parameters gracefully
- Maintains consistency with project architecture and coding standards
- Zero breaking changes - existing functionality unaffected

### CODE REVIEW

File: test/OpenChat.PlaygroundApp.Tests/Options/UpstageArgumentOptionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+        baseOptions.ShouldBeOfType<UpstageArgumentOptions>();
+        baseOptions.Help.ShouldBeFalse(); // Can access inherited properties
+
+        // Cast to specific type to access Upstage-specific properties
+        var upstageOptions = baseOptions as UpstageArgumentOptions;
+        upstageOptions.ShouldNotBeNull();
+        upstageOptions!.BaseUrl.ShouldBeNull();
+        upstageOptions.ApiKey.ShouldBeNull();
+        upstageOptions.Model.ShouldBeNull();
+    }
```
[the remark this answers is not in the document]
Comment (gremh97) [reply test/OpenChat.PlaygroundApp.Tests/Options/UpstageArgumentOptionsTests.cs]: 넵 revert 했습니다. 어디서 얼핏 상속테스트를 본거 같아서 넣었는데 여기가 아니였나보네요
